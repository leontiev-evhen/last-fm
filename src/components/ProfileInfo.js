import React from 'react';
import PropTypes from 'prop-types';
import { Image } from './Image';
import { Tag } from './Tag';

export function ProfileInfo({ profile }) {
	return (
		<div className="profile-section">
			<div className="row">
				<div className="col">
					<Image
						path={
							profile.image.length > 3 ? profile.image[3]['#text'] : undefined
						}
					/>
				</div>
				<div className="col">
					<div className="profile-info">
						<h2>{profile.name}</h2>
						<div className="row">
							<h2>Biography</h2>
							<div
								className="profile-content"
								dangerouslySetInnerHTML={{ __html: profile.bio.content }}
							/>
						</div>
					</div>
				</div>
				<div className="col">
					<div className="profile-info">
						<h2>Tags</h2>
						{profile.tags.tag.map((tag, index) => (
							<div key={index}>
								<Tag name={tag.name} />
							</div>
						))}
					</div>
				</div>
			</div>
		</div>
	);
}

ProfileInfo.propTypes = {
	profile: PropTypes.shape({
		name: PropTypes.string.isRequired,
		image: PropTypes.array,
		bio: PropTypes.shape({
			content: PropTypes.string.isRequired,
		}).isRequired,
		tags: PropTypes.shape({
			tag: PropTypes.arrayOf(PropTypes.object).isRequired,
		}).isRequired,
	}).isRequired,
};
