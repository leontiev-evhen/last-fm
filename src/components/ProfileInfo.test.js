import React from 'react';
import { shallow } from 'enzyme';
import { ProfileInfo } from './ProfileInfo';

describe('ProfileInfo component', () => {
  const props = {
    profile: {
      name: 'profileName',
      image: [{
        '#text': 'image1',
      }, {
        '#text': 'image2',
      }, {
        '#text': 'image3',
      }, {
        '#text': 'image3',
      }],
      bio: {
        content: 'content...',
      },
      tags: {
        tag: [{
          name: 'tag1'
        }, {
          name: 'tag2'
        }, {
          name: 'tag3'
        }],
      }
    },
  };
  const component = shallow(<ProfileInfo  {...props} />);

  it('render correctly ProfileInfo component', () => {
    expect(component).toMatchSnapshot();
  });
  it('correct profile name', () => {
     expect(component.find('h2').first().text()).toEqual(props.profile.name)
  });
  it('correct Image src', () => {
    expect(component.find('Image').prop('path')).toEqual(props.profile.image[3]['#text']);
  });
  it('correct profile content', () => {
    expect(component.find('.profile-content').prop('dangerouslySetInnerHTML')['__html']).toEqual(props.profile.bio.content);
  });
  it('correct profile tags', () => {
    expect(component.find('Tag')).toHaveLength(props.profile.tags.tag.length);
  });
  describe('Image array size is less than three', () => {
    const nextProps = {
      profile: {
        ...props.profile,
        image: [{
          '#text': 'image1',
        }, {
          '#text': 'image2',
        }, {
          '#text': 'image3',
        }],
      },
    };
    const component = shallow(<ProfileInfo {...nextProps} />);
    it('Image component path is undefined', () => {
      expect(component.find('Image').prop('path')).toEqual(undefined);
    });
  });
});
