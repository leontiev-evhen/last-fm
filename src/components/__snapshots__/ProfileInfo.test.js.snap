// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo component render correctly ProfileInfo component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileInfo
    profile={
      Object {
        "bio": Object {
          "content": "content...",
        },
        "image": Array [
          Object {
            "#text": "image1",
          },
          Object {
            "#text": "image2",
          },
          Object {
            "#text": "image3",
          },
          Object {
            "#text": "image3",
          },
        ],
        "name": "profileName",
        "tags": Object {
          "tag": Array [
            Object {
              "name": "tag1",
            },
            Object {
              "name": "tag2",
            },
            Object {
              "name": "tag3",
            },
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col"
        >
          <Image
            path="image3"
          />
        </div>
        <div
          className="col"
        >
          <div
            className="profile-info"
          >
            <h2>
              profileName
            </h2>
            <div
              className="row"
            >
              <h2>
                Biography
              </h2>
              <div
                className="profile-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "content...",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="profile-info"
          >
            <h2>
              Tags
            </h2>
            <div>
              <Tag
                name="tag1"
              />
            </div>
            <div>
              <Tag
                name="tag2"
              />
            </div>
            <div>
              <Tag
                name="tag3"
              />
            </div>
          </div>
        </div>
      </div>,
      "className": "profile-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col"
          >
            <Image
              path="image3"
            />
          </div>,
          <div
            className="col"
          >
            <div
              className="profile-info"
            >
              <h2>
                profileName
              </h2>
              <div
                className="row"
              >
                <h2>
                  Biography
                </h2>
                <div
                  className="profile-content"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "content...",
                    }
                  }
                />
              </div>
            </div>
          </div>,
          <div
            className="col"
          >
            <div
              className="profile-info"
            >
              <h2>
                Tags
              </h2>
              <div>
                <Tag
                  name="tag1"
                />
              </div>
              <div>
                <Tag
                  name="tag2"
                />
              </div>
              <div>
                <Tag
                  name="tag3"
                />
              </div>
            </div>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Image
              path="image3"
            />,
            "className": "col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "path": "image3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="profile-info"
            >
              <h2>
                profileName
              </h2>
              <div
                className="row"
              >
                <h2>
                  Biography
                </h2>
                <div
                  className="profile-content"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "content...",
                    }
                  }
                />
              </div>
            </div>,
            "className": "col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  profileName
                </h2>,
                <div
                  className="row"
                >
                  <h2>
                    Biography
                  </h2>
                  <div
                    className="profile-content"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "content...",
                      }
                    }
                  />
                </div>,
              ],
              "className": "profile-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "profileName",
                },
                "ref": null,
                "rendered": "profileName",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2>
                      Biography
                    </h2>,
                    <div
                      className="profile-content"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "content...",
                        }
                      }
                    />,
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Biography",
                    },
                    "ref": null,
                    "rendered": "Biography",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "profile-content",
                      "dangerouslySetInnerHTML": Object {
                        "__html": "content...",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="profile-info"
            >
              <h2>
                Tags
              </h2>
              <div>
                <Tag
                  name="tag1"
                />
              </div>
              <div>
                <Tag
                  name="tag2"
                />
              </div>
              <div>
                <Tag
                  name="tag3"
                />
              </div>
            </div>,
            "className": "col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Tags
                </h2>,
                Array [
                  <div>
                    <Tag
                      name="tag1"
                    />
                  </div>,
                  <div>
                    <Tag
                      name="tag2"
                    />
                  </div>,
                  <div>
                    <Tag
                      name="tag3"
                    />
                  </div>,
                ],
              ],
              "className": "profile-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tags",
                },
                "ref": null,
                "rendered": "Tags",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <Tag
                    name="tag1"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "tag1",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <Tag
                    name="tag2"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "tag2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": <Tag
                    name="tag3"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "tag3",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col"
          >
            <Image
              path="image3"
            />
          </div>
          <div
            className="col"
          >
            <div
              className="profile-info"
            >
              <h2>
                profileName
              </h2>
              <div
                className="row"
              >
                <h2>
                  Biography
                </h2>
                <div
                  className="profile-content"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "content...",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="profile-info"
            >
              <h2>
                Tags
              </h2>
              <div>
                <Tag
                  name="tag1"
                />
              </div>
              <div>
                <Tag
                  name="tag2"
                />
              </div>
              <div>
                <Tag
                  name="tag3"
                />
              </div>
            </div>
          </div>
        </div>,
        "className": "profile-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col"
            >
              <Image
                path="image3"
              />
            </div>,
            <div
              className="col"
            >
              <div
                className="profile-info"
              >
                <h2>
                  profileName
                </h2>
                <div
                  className="row"
                >
                  <h2>
                    Biography
                  </h2>
                  <div
                    className="profile-content"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "content...",
                      }
                    }
                  />
                </div>
              </div>
            </div>,
            <div
              className="col"
            >
              <div
                className="profile-info"
              >
                <h2>
                  Tags
                </h2>
                <div>
                  <Tag
                    name="tag1"
                  />
                </div>
                <div>
                  <Tag
                    name="tag2"
                  />
                </div>
                <div>
                  <Tag
                    name="tag3"
                  />
                </div>
              </div>
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Image
                path="image3"
              />,
              "className": "col",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "path": "image3",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="profile-info"
              >
                <h2>
                  profileName
                </h2>
                <div
                  className="row"
                >
                  <h2>
                    Biography
                  </h2>
                  <div
                    className="profile-content"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "content...",
                      }
                    }
                  />
                </div>
              </div>,
              "className": "col",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    profileName
                  </h2>,
                  <div
                    className="row"
                  >
                    <h2>
                      Biography
                    </h2>
                    <div
                      className="profile-content"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "content...",
                        }
                      }
                    />
                  </div>,
                ],
                "className": "profile-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "profileName",
                  },
                  "ref": null,
                  "rendered": "profileName",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h2>
                        Biography
                      </h2>,
                      <div
                        className="profile-content"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "content...",
                          }
                        }
                      />,
                    ],
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Biography",
                      },
                      "ref": null,
                      "rendered": "Biography",
                      "type": "h2",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "profile-content",
                        "dangerouslySetInnerHTML": Object {
                          "__html": "content...",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="profile-info"
              >
                <h2>
                  Tags
                </h2>
                <div>
                  <Tag
                    name="tag1"
                  />
                </div>
                <div>
                  <Tag
                    name="tag2"
                  />
                </div>
                <div>
                  <Tag
                    name="tag3"
                  />
                </div>
              </div>,
              "className": "col",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Tags
                  </h2>,
                  Array [
                    <div>
                      <Tag
                        name="tag1"
                      />
                    </div>,
                    <div>
                      <Tag
                        name="tag2"
                      />
                    </div>,
                    <div>
                      <Tag
                        name="tag3"
                      />
                    </div>,
                  ],
                ],
                "className": "profile-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Tags",
                  },
                  "ref": null,
                  "rendered": "Tags",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Tag
                      name="tag1"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "tag1",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Tag
                      name="tag2"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "tag2",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Tag
                      name="tag3"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "tag3",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
